### PR TITLE
Deployment documentation

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: docs/deployment.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Deployment
+
+## Deploying locally
+
+Your function can run locally either by starting it through Visual
```
Comment (amanda-tarafa) [inline docs/deployment.md:5]: If it depends on the Invoker or a custom invoker, seems evident to us but might not be to everyone.

File: docs/deployment.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+means you can run functions on Cloud Run with ease.
+
+There are detailed instructions for C# available as part of the
+Cloud Run documentation:
+
+- [Quickstart for C#](https://cloud.google.com/run/docs/quickstarts/build-and-deploy#c)
+- [Codelab for C#](https://codelabs.developers.google.com/codelabs/cloud-run-hello-csharp/index.html)
+
+The instructions apply to functions in exactly the same way as
+normal, with the same Dockerfile configuration. The `PORT`
```
Comment (amanda-tarafa) [inline docs/deployment.md:27]: This sentence is weird: in exactly the same way as a normal, ?

File: docs/deployment.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+environment variable in your Dockerfile, with a line like this:
+
+```text
+ENV FUNCTION_TARGET HelloFunctions.Function
+```
+
+## Deploying to Google Cloud Functions
+
+> **NOTE:**  
+> The `dotnet3` runtime is currently available on an "allow
```
Comment (amanda-tarafa) [inline docs/deployment.md:39]: If there are instructions on how to get whitelisted, or at least a note on who gets whitelisted, we can avoid people asking.

File: docs/deployment.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+a function written in every language. The .NET-specific aspects are:
+
+- The runtime should be specified as `dotnet3`
+- The entry point should be specified as the name of the function
+  type, including namespace (e.g. `HelloFunctions.Function`)
+
+The command line options are used to specify how the function is
+triggered.
+
+### HTTP functions
```
Comment (amanda-tarafa) [inline docs/deployment.md:58]: No section for Cloud Events?

**Comment by amanda-tarafa** [comment]
A couple of my comments here had already been replied to in the bigger PR so ignore here.

File: docs/deployment.md (hunk 3 above)
Comment (grant) [reply docs/deployment.md:39]: +1. This is a bit ominous.

File: docs/deployment.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+**Sample triggers**
+
+Replace any ***bold italic*** parts with your project ID, bucket name and so forth.
+
+Trigger type             | Payload type   | Sample command line options
+------------------------ | -------------- | --------------------------
+Cloud Storage operation  | StorageObject  | --trigger-event google.storage.object.finalize --trigger-resource ***my-gcs-bucket***
+Pub/Sub message          | PubSubMessage  | --trigger-topic ***my-pubsub-topic-id***
+Firestore event          | FirestoreEvent | --trigger event providers/cloud.firestore/eventTypes/document.write --trigger-resource 'projects/***my-project***/databases/(default)/documents/***my-collection***/{document}'
```
Comment (grant) [inline docs/deployment.md:84]: Is there a dash missing? `--trigger event `.

File: docs/deployment.md (hunk 1 above)
Comment (jskeet) [reply docs/deployment.md]: Done.

File: docs/deployment.md (hunk 2 above)
Comment (jskeet) [reply docs/deployment.md]: Fixed - but check that it's clear :)

File: docs/deployment.md (hunk 3 above)
Comment (jskeet) [reply docs/deployment.md:47]: Added a bit here. PTAL.

File: docs/deployment.md (hunk 4 above)
Comment (jskeet) [reply docs/deployment.md:69]: Renamed "legacy event" to "Cloud Event" now that that's all we have.

File: docs/deployment.md (hunk 5 above)
Comment (jskeet) [reply docs/deployment.md]: Well spotted - fixed, thanks.

**Pull request merged**